### PR TITLE
fix server IP in info panel

### LINKED ISSUE #416: Wrong IP address in info panel
Version: 2.7.4

### Bug Description
obrazek

I tested standalone Tracy on Vagrant / Laravel Homestead.

### Expected Behavior
obrazek

### Possible Solution
revert commit ce31ae5d5d92709d8624b4ee0f46cd9f16aa7c1e

### PR DESCRIPTION
**Pull request by petrparolek**
closes #416

- bug fix
- BC break? no

- doc PR: nette/docs will be

### CODE REVIEW

File: src/Tracy/Debugger/Debugger.php [hunk 1]
```diff
@@ -25,9 +25,14 @@ class Debugger
 		PRODUCTION = true,
 		DETECT = null;
 
+	/** @var bool */
+	public const BEHIND_PROXY = false;
```
Comment (janbarasek) [inline src/Tracy/Debugger/Debugger.php:29]: @petrparolek The constant is not used anywhere.

File: src/Tracy/Debugger/Debugger.php [hunk 2]
```diff
@@ -25,9 +25,14 @@ class Debugger
...
+
 	public const COOKIE_SECRET = 'tracy-debug';
 
-	/** @var bool in production mode is suppressed any debugging output */
+    public static $behindProxy = false;
```
Comment (janbarasek) [inline src/Tracy/Debugger/Debugger.php:33]: @petrparolek Fix codestyle, please.

File: src/Tracy/Bar/panels/info.panel.phtml [hunk 3]
```diff
@@ -32,6 +32,13 @@ $ipFormatter = static function (?string $ip): ?string {
 	return $ip;
 };
 
+if(Debugger::$behindProxy) {
```
Comment (janbarasek) [inline src/Tracy/Bar/panels/info.panel.phtml:35]: @petrparolek codestyle: `if (Debugger::$behindProxy === true) {`

File: src/Tracy/Bar/panels/info.panel.phtml [hunk 4]
```diff
@@ -32,6 +32,13 @@ $ipFormatter = static function (?string $ip): ?string {
...
+    $serverIP = $ipFormatter(gethostbyname($serverName = php_uname('n')));
+} else {
+    $serverIP = $ipFormatter($serverIp = $_SERVER['SERVER_ADDR'] ?? null);
```
Comment (janbarasek) [inline src/Tracy/Bar/panels/info.panel.phtml:38]: @petrparolek Why `$serverIP` is not `$serverIp`?

**Comment by petrparolek** [comment]
@dg  why there a lo of your commits?

**Comment by janbarasek** [comment]
@petrparolek Please merge current master to your branch. Your branch contains 47 changed files. There are so many...

**Comment by petrparolek** [comment]
I don't know why @dg  force pushed to master!

**Comment by janbarasek** [comment]
@petrparolek Please be polite to David. You want merge your changes, you must fix conflict! :)

**Comment by petrparolek** [comment]
I made commit again

**Comment by janbarasek** [comment]
@petrparolek Thanks, but please fix codestyle and tests first.

File: src/Tracy/Bar/panels/info.panel.phtml [hunk 5]
```diff
@@ -32,6 +32,13 @@ $ipFormatter = static function (?string $ip): ?string {
 	return $ip;
 };
 
+if(Debugger::$behindProxy === true) {
```
Comment (janbarasek) [inline src/Tracy/Bar/panels/info.panel.phtml:35]: Please fix codestyle: `if (Debugger::$behindProxy === true) {` (space between `if` and bracket).

File: src/Tracy/Bar/panels/info.panel.phtml [hunk 6]
```diff
@@ -32,6 +32,13 @@ $ipFormatter = static function (?string $ip): ?string {
...
+    $serverIp = $ipFormatter(gethostbyname($serverName = php_uname('n')));
+} else {
+    $serverIp = $ipFormatter($serverIp = $_SERVER['SERVER_ADDR'] ?? null);
+    $serverName = $serverIp !== null && $serverIp !== '127.0.0.1' && $serverIp !== '::1' ? gethostbyaddr($serverIp) : 'localhost';
```
Comment (janbarasek) [inline src/Tracy/Bar/panels/info.panel.phtml:39]: Please use tabs, no spaces.

**Comment by petrparolek** [comment]
tests are falling because of last commit from @dg

File: src/Tracy/Bar/panels/info.panel.phtml (hunk 6 above)
Comment (petrparolek) [reply src/Tracy/Bar/panels/info.panel.phtml]: thx

File: src/Tracy/Bar/panels/info.panel.phtml (hunk 5 above)
Comment (petrparolek) [reply src/Tracy/Bar/panels/info.panel.phtml]: thx

**Comment by janbarasek** [comment]
@petrparolek Ok, thanks. Who define value of public property `$behindProxy`?

Can you implement autodetection?

What happen when real user IP (before proxy) is fake?

**Comment by petrparolek** [comment]
I think autodection is not possible, in default mode tracy show IP address as in version 2.6.3 and user can switch to your solution by `Debugger::$behindProxy = true;`, if @dg agree with solution, we can add `Debugger::$behindProxy = true;` to doc and to DI extension

**Comment by dg** [comment]
Sorry for force push. 

Hey boys we are talking about a totally irrelevant thing hidden somewhere in the panel. Spoofing doen't matter, you would have to spoof yourself. Adding a new configuration variable to the Debugger is an inadequate solution. 

I suggest to removing this item from the info panel :-)

**Comment by petrparolek** [comment]
@dg  I suggest to revert comnmit  #ce31ae5 and close this PR.

**Pull request closed without merging**